### PR TITLE
[opencv4] Update to 4.7.0.

### LINKED ISSUE #28807: [opencv] update to 4.7.0
Library name: OpenCV

New version number: 4.7.0

Released on 2022-12-29. [Release notes](https://github.com/opencv/opencv/wiki/ChangeLog#version470)

### PR DESCRIPTION
**Pull request by Sibras**
[opencv4] Update to 4.7.0.

Fixes #28807

### CODE REVIEW

**Comment by cnSchwarzer** [comment]
Duplicates #28434

**Comment by Sibras** [comment]
> Duplicates #28434 I also updated all ports blocking ffmpeg 5.1.2 in that PR.

Yep, ive seen that.
This is so that it can be updated/tested standalone without the ffmpeg changes (and all the others) and hopefully get merged faster.

**Comment by cnSchwarzer** [comment]
> > Duplicates #28434 I also updated all ports blocking ffmpeg 5.1.2 in that PR.
> 
> 
> 
> Yep, ive seen that.
> 
> This is so that it can be updated/tested standalone without the ffmpeg changes (and all the others) and hopefully get merged faster.

Could you help fan out all ports? Much thanks! I didn't fan out because I need to test whether ffmpeg can pass CI

**Comment by Sibras** [comment]
> > > Duplicates #28434 I also updated all ports blocking ffmpeg 5.1.2 in that PR.
> > 
> > 
> > Yep, ive seen that.
> > This is so that it can be updated/tested standalone without the ffmpeg changes (and all the others) and hopefully get merged faster.
> 
> Could you help fan out all ports? Much thanks! I didn't fan out because I need to test whether ffmpeg can pass CI

Yep, done that now.

File: ports/rtabmap/portfile.cmake [hunk 1]
```diff
@@ -1,14 +1,16 @@
+vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}
+
```
Comment (JonLiu1993) [inline ports/rtabmap/portfile.cmake:2]:
```suggestion
```

Comment (JonLiu1993) [reply ports/rtabmap/portfile.cmake:2]: @Sibras, Thanks for your pr, later we no longer need to add this line of code when using `v${VERSION}`

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
Requires PR https://github.com/microsoft/vcpkg/pull/28954

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
@Sibras, Thanks for your PR, you forgot to modify the portfile.cmake file of opencv.

**Comment by Sibras** [comment]
> @Sibras, Thanks for your PR, you forgot to modify the portfile.cmake file of opencv.

The portfile.cmake file for opencv is an empty package, what needs to be modified in it?

**Comment by JonLiu1993** [comment]
> > @Sibras, Thanks for your PR, you forgot to modify the portfile.cmake file of opencv.
> 
> The portfile.cmake file for opencv is an empty package, what needs to be modified in it?

My mistake, I didn't open it to see, the review found that there seems to be a portfile.cmake file missing from the modification.

**Comment by BillyONeal** [comment]
Thanks for the update! /cc @cenit

**Pull request merged**